File: 백준/Silver/11053. 가장 긴 증가하는 부분 수열/가장 긴 증가하는 부분 수열.py
from sys import *
from collections import * 
from heapq import * 
from functools import *
from math import *
from itertools import *

#setrecursionlimit(10**9)
input = stdin.readline


n = int(input())

arr = list(map(int,input().split()))

dp = [1] * 1001


ans = 1

for i in range(n):
    for j in range(i+1):
        if arr[i] > arr[j]:
            dp[i] = max(dp[i],dp[j]+1)
    
    
    ans = max(ans,dp[i])

print(ans)
    

File: 백준/Silver/11053. 가장 긴 증가하는 부분 수열/가장 긴 증가하는 부분 수열.swift
let n = Int(readLine()!)!

let arr = readLine()!.split{$0 == " "}.map{Int($0)!}


var dp: [Int] = [Int](repeating: 1, count: n)

for i in 0..<n {
    for j in 0...i {
        
        if arr[i] > arr[j] {
            dp[i] = max(dp[i],dp[j]+1)
        }
    }
}

print(dp.max()!)
